autoload/go/test.vim

```diff
@@ -242,108 +240,141 @@ function! s:show_errors(args, exit_val, messages) abort
   endif
 
   if a:args.winnr == winnr()
-    call go#list#Populate(l:listtype, errors, join(a:args.cmd))
     call go#list#Window(l:listtype, len(errors))
     if !empty(errors) && !a:args.bang
       call go#list#JumpToFirst(l:listtype)
     endif
   endif
 endfunction
 
-function! s:parse_errors(lines) abort
-  let errors = []
-  let paniced = 0 " signals whether all remaining lines should be included in errors.
-  let test = ''
-
-  " NOTE(arslan): once we get JSON output everything will be easier :)
-  " https://github.com/golang/go/issues/2981
```

Comment:
Can you add this back somewhere in the code. I'm still looking for the JSON support :)